migrations/20210612_initial.sql

```diff
@@ -0,0 +1,104 @@
+CREATE TABLE profiles (
+  profile_id INTEGER NOT NULL PRIMARY KEY,
+  display_name TEXT NOT NULL,
+  properties TEXT NOT NULL -- JSON with user or contact profile
+);
+
+CREATE TABLE known_servers(
+  host TEXT NOT NULL,
+  port TEXT NOT NULL,
+  key_hash BLOB,
+  PRIMARY KEY (host, port)
+) WITHOUT ROWID;
+
+CREATE TABLE contacts (
+  contact_id INTEGER NOT NULL PRIMARY KEY,
+  display_name TEXT NOT NULL, -- display name set by local user
+  properties TEXT NOT NULL, -- JSON set by local user
+  profile_id INTEGER REFERENCES profiles, -- sent by remote contact
+  contact_status TEXT NOT NULL DEFAULT ''
```

Comment:
let's not have empty strings as default please - it's no better than `NULL` except for when you have to work with it in the db - then in many interfaces it's unclear whether you're dealing with an empty string or a null

Reply:
I've made the same in introductions/invitations... we should change it there too then? If I read it into Text/ByteString  - will NULL be read as an empty string?

Reply:
>  If I read it into Text/ByteString - will NULL be read as an empty string?

not sure, I can make a test later - but you still have to treat either a NULL or an empty string somewhere. Having it as empty string only moves validation further from database